### ISSUE TITLE
Meshery Operator to watch Broker custom resource for changes to Endpoint location

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery-operator#333]
#### Current Behavior
Meshery Server connects to Meshery Broker by way of the information contained in the `broker` custom resource,  specifically, the IP / hostname contained in the `Endpoint` section.

```
...
....
Status:
  Endpoint:
    External:  localhost:4222
    Internal:  10.96.168.204:4222
Events:        <none>
```

#### Desired Behavior
When this endpoint location changes, Meshery Server learns of it and updates the location it connects to.

#### Implementation
1. Ideally, Meshery Operator informs Meshery Server of this change (pushes the change), so that this is an event-based flow.2. 
1. If this isn't feasible, users should be able to manually have Meshery Server refresh this information. Let's avoid period polling initially as it only brings in more concerns.

#### Acceptance Tests
1. Users can update networking of their broker, update the endpoint location in their custom resource, and then have Meshery Server refresh the endpoint location information. Then, establish a new subscription to the MeshSync topic.2. 

#### Mockups
See Meshery Architecture deck for sequence diagrams.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

**Comment by siddhikhapare** [comment]
@leecalcote  I would like to work on this issue. Can you give me some pointer on this?

**Comment by acald-creator** [comment]
Bump

I noticed this happening, and I am researching to see how to make this better.

**Comment by rohan472000** [comment]
can i take this issue?

**Comment by leecalcote** [comment]
@rohan472000, yes, please propose a solution so that your approach can be vetted before you spend time implementing.

**Comment by rohan472000** [comment]
@leecalcote we can implement a watch on the custom resource, which tracks changes to the endpoint location, 
the meshery operator can trigger this watch, which will notify Meshery Server of any changes to the 
endpoint location.

or

we can provide a manual refresh option to Meshery Server, this option will allow users to update the endpoint location information by triggering a refresh action. Once the endpoint location changes, Meshery Server should establish a new connection to the updated location.

or 

we can use API based approach, that Meshery Server can use to retrieve the current endpoint location. Whenever the endpoint location changes, the API can return the updated information to Meshery Server. Server can call this API periodically to fetch endpoint location.

**Comment by i-dipanshu** [comment]
Hey @leecalcote and @acald-creator, is this issue still relevant ?

**Comment by leecalcote** [comment]
@i-dipanshu yes it is.

**Comment by i-dipanshu** [comment]
Thank you, @leecalcote , for the update on the issue. I've started my research to understand and resolve it. While I have experience with few other Kubernetes operators, Meshery operator is new to me. Any specific insights to help resolve this issue would be appreciated.

**Comment by abohoss** [comment]
Is this issue still open?

**Comment by Om-Beast** [comment]
Hi! I noticed this issue was recently referenced from #786 during the broker modernization work.

Before I start investigating, could you confirm whether this issue is still outstanding after the recent networking and BrokerSpec changes, or if only a specific part of the original behavior remains to be implemented? If it's still open, I'd be happy to work on the remaining scope.
